app/graphql/types/customers/billing_configuration.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Types
+  module Customers
+    class BillingConfiguration < Types::BaseObject
+      graphql_name 'CustomerBillingConfiguration'
+
+      field :id, ID, null: false
```

Comment:
What is this ID? Not sure to understand where it comes from :thinking:

Reply:
this ID is required by the frontend 😕

Thing is, our cache management system uses IDs to manage it's cache updates.

When we define a new type, the cache systems sees those resources as separate models, hence warning us about potential data loss.

To prevent this, I asked Lovro to allow us to request an ID on the type level. Also, note this ID is not present in the input type.

This ID could have the value of the parent (here customer) ID in case having a random one is an issue.
Also, there is no need for this ID to be stored in DB as it's only used bu the cache management system to properly store and update a resource.

Does it make sense and answer your question ?

Reply:
Alright! Thank you for the explanation :)